core/trino-main/src/main/java/io/trino/sql/planner/planprinter/CounterBasedAnonymizer.java

```diff
@@ -0,0 +1,354 @@
+import io.trino.sql.tree.ExpressionRewriter;
+import io.trino.sql.tree.ExpressionTreeRewriter;
+import io.trino.sql.tree.GenericLiteral;
+import io.trino.sql.tree.IntervalLiteral;
+import io.trino.sql.tree.Literal;
+import io.trino.sql.tree.LongLiteral;
+import io.trino.sql.tree.NullLiteral;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.sql.tree.SymbolReference;
+import io.trino.sql.tree.TimeLiteral;
+import io.trino.sql.tree.TimestampLiteral;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.trino.sql.planner.Partitioning.ArgumentBinding;
+import static io.trino.sql.planner.plan.StatisticsWriterNode.WriteStatisticsHandle;
+import static io.trino.sql.planner.plan.StatisticsWriterNode.WriteStatisticsTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.CreateTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.DeleteTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.InsertTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.MergeTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.RefreshMaterializedViewTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.TableExecuteTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.UpdateTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.WriterTarget;
+import static io.trino.sql.tree.ExpressionTreeRewriter.rewriteWith;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Locale.ENGLISH;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * An implementation of {@link Anonymizer} which uses simple counter-based mechanism
+ */
+public class CounterBasedAnonymizer
+        implements Anonymizer
+{
+    private enum ObjectType
+    {
+        CATALOG,
+        SCHEMA,
+        TABLE,
+        COLUMN,
+        SYMBOL,
+        LITERAL,
+        VALUE
+    }
+
+    private final Map<String, String> anonymizedMap = new HashMap<>();
+    private final Map<ObjectType, Integer> counterMap = Arrays.stream(ObjectType.values())
+            .collect(toMap(objectType -> objectType, objectType -> 0));
+
+    @Override
+    public String anonymize(Type type, String value)
+    {
+        String formattedTypeName = type.getDisplayName()
+                .replaceAll("\\W", "_")
+                .replaceAll("__", "_");
+        return formattedTypeName + "_" + anonymize(value, ObjectType.VALUE);
+    }
+
+    @Override
+    public String anonymize(Symbol symbol)
+    {
+        return anonymize(symbol.getName(), ObjectType.SYMBOL);
+    }
+
+    @Override
+    public String anonymizeColumn(String symbol)
+    {
+        return anonymize(symbol, ObjectType.COLUMN);
+    }
+
+    @Override
+    public String anonymize(Expression expression)
+    {
+        return anonymizeExpression(expression).toString();
+    }
+
+    private Expression anonymizeExpression(Expression expression)
```

Comment:
I still have concerns about this, which I mentioned before in a related context. Anonymized expressions are not valid SQL, so we should not be trying to construct an AST out of them (effectively what this method does)

Reply:
The API of Anonymizer is `String anonymize(Expression expression)`, so we don't really want an Anonymized expression, an anonymised string representation of the Expression will do.
This method is creating an Expression because it was convenient to use an ExpressionRewriter to anonymise literals and then use Expression#toString on the result.
I think an alternative could be that write something similar to `ExpressionFormatter#Formatter` that delegates to existing formatter for all methods except the ones we want to anonymise (visitXXXLiteral).
Would that be better or is there another way that you would recommend instead ?

Reply:
That would be better. Alternatively, we should consider and explore making the expression formatter itself anonymizer-aware.

Reply:
I've updated the expression formatter such that we could use that directly instead of creating anonymized AST. PTAL @martint